### PR TITLE
sql.Builder Join include Querier's error

### LINKED ISSUE #1529: Selector should gather all errors or propagate inner error
- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/facebook/ent/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯

```go
func ApplyQuery(node *sqlgraph.Node, query string) func(s *sql.Selector) {
  return func(s *sql.Selector) {
    s.Where(sql.P(func(b *sql.Builder) {
            // this error is ignored - should just panic ?
	    b.AddError(errors.New("ERR"))
  }))
}
```

The error in Builder will just ignored.

In `func (b *Builder) join(qs []Querier, sep string) *Builder` error is ignored,
so in `func (s *Selector) Query() (string, []interface{})` all error is ignored.

Maybe `Builder.Join` also collect Querier's error ?

## Expected Behavior 🤔

select error out

## Your Environment 🌎

| Tech        | Version |
| ----------- | ------- |
| Go          | 1.16  |
| Ent         | 0.7   |
| Database    | -   |
| Driver      | -     |

### PR DESCRIPTION
**Pull request by wenerme**
fix #1529

### CODE REVIEW

File: dialect/sql/builder.go [hunk 1]
```diff
@@ -2488,6 +2497,9 @@ func (b *Builder) join(qs []Querier, sep string) *Builder {
 		b.WriteString(query)
 		b.args = append(b.args, args...)
 		b.total += len(args)
+		if qe, ok := q.(querierErr); ok {
+			b.AddError(qe.Err())
+		}
```
Comment (a8m) [inline dialect/sql/builder.go:2502]: No need to call this method if `Err()` is nil. 
```suggestion
		if qe, ok := q.(querierErr); ok {
			if err := qe.Err(); err != nil {
				b.AddError(err)
			}
		}
```

**Review by a8m — APPROVED** [review]
Looks good. Thanks for adding this.

File: dialect/sql/builder.go (hunk 1 above)
Comment (wenerme) [reply dialect/sql/builder.go:2502]: this pr makes AddError ignored nil error, makes code more clear, or do you thing all AddError should be checked ?

Comment (a8m) [reply dialect/sql/builder.go:2502]: We can keep the logic in `AddError` that ignores nil errors, but there's no need to call `AddError` on every `join` call if it's unnecessary. Sounds good?

Comment (wenerme) [reply dialect/sql/builder.go:2502]: Sounds good.

**Pull request merged**